pkg/scheduler/algorithm/priorities/balanced_resource_allocation_test.go

```diff
@@ -402,19 +402,29 @@ func TestBalancedResourceAllocation(t *testing.T) {
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			nodeNameToInfo := schedulernodeinfo.CreateNodeNameToInfoMap(test.pods, test.nodes)
-			if len(test.pod.Spec.Volumes) > 0 {
-				maxVolumes := 5
-				for _, info := range nodeNameToInfo {
-					info.TransientInfo.TransNodeInfo.AllocatableVolumesCount = getExistingVolumeCountForNode(info.Pods(), maxVolumes)
-					info.TransientInfo.TransNodeInfo.RequestedVolumes = len(test.pod.Spec.Volumes)
+
+			for _, hasMeta := range []bool{true, false} {
+				var metadata *priorityMetadata
+				if hasMeta {
+					metadata = &priorityMetadata{
+						nonZeroRequest: getNonZeroRequests(test.pod),
+					}
+				}
+
+				if len(test.pod.Spec.Volumes) > 0 {
+					maxVolumes := 5
+					for _, info := range nodeNameToInfo {
+						info.TransientInfo.TransNodeInfo.AllocatableVolumesCount = getExistingVolumeCountForNode(info.Pods(), maxVolumes)
+						info.TransientInfo.TransNodeInfo.RequestedVolumes = len(test.pod.Spec.Volumes)
+					}
+				}
+				list, err := priorityFunction(BalancedResourceAllocationMap, nil, metadata)(test.pod, nodeNameToInfo, test.nodes)
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+				}
+				if !reflect.DeepEqual(test.expectedList, list) {
+					t.Errorf("expected %#v, got %#v", test.expectedList, list)
 				}
-			}
-			list, err := priorityFunction(BalancedResourceAllocationMap, nil, nil)(test.pod, nodeNameToInfo, test.nodes)
```

Comment:
we are passing nil already in the test, was the test crashing all the time?

Reply:
Let me try to explain here: "non-typed" nil and "typed" nil (i.e. "var foo *meta") behave differently in the type assertion.

Current testcase and code base are always using "non-typed" nil, so it works fine all the time. In this PR, @draveness introduced a _potential_ "typed" nil (when hasMeta=false), so it can pass type assertion. It's a Golang gotcha which IMO should be aware of, but shouldn't practice as a pattern in code.

@draveness my suggestion is:

1. avoid introducing "typed" nil as it can confuse people
1. eliminate the `&& meta != nil` check in this PR as well as #77498 

Thoughts?

BTW: a write-up on [typed nil](https://dave.cheney.net/2017/08/09/typed-nils-in-go-2).

Reply:
SGTM, updated as your suggestion.

Reply:
sure, but in the new code, if you pass a typed nil, wouldn't that cause a crash? shouldn't we make sure that does not happen?